src/SmartTransactionsController.ts

```diff
@@ -364,41 +434,40 @@
   #updateSmartTransaction(
     smartTransaction: SmartTransaction,
     {
-      chainId = this.config.chainId,
+      chainId = this.#chainId,
     }: {
       chainId: Hex;
     },
   ) {
-    const { smartTransactionsState } = this.state;
-    const { smartTransactions } = smartTransactionsState;
+    const {
+      smartTransactionsState: { smartTransactions },
+    } = this.state;
     const currentSmartTransactions = smartTransactions[chainId] ?? [];
     const currentIndex = currentSmartTransactions?.findIndex(
       (stx) => stx.uuid === smartTransaction.uuid,
     );
+
     if (currentIndex === -1) {
       return; // Smart transaction not found, don't update anything.
     }
-    this.update({
-      smartTransactionsState: {
-        ...smartTransactionsState,
-        smartTransactions: {
-          ...smartTransactionsState.smartTransactions,
-          [chainId]: smartTransactionsState.smartTransactions[chainId].map(
-            (existingSmartTransaction, index) => {
-              return index === currentIndex
-                ? { ...existingSmartTransaction, ...smartTransaction }
-                : existingSmartTransaction;
-            },
-          ),
-        },
-      },
+
+    this.update((state) => {
+      const currentTransaction = cloneDeep(
+        state.smartTransactionsState.smartTransactions[chainId][currentIndex],
+      );
+      if (
+        [chainId, currentIndex].every((key) => isSafeDynamicKey(String(key)))
+      ) {
+        state.smartTransactionsState.smartTransactions[chainId][currentIndex] =
```

Comment:
Hmm looks like this fix didn't get rid of the alert. Could we try something more explicit e.g.

```
if (isSafeDynamicKey(chainId) && isSafeDynamicKey(String(currentIndex))) {
```

Reply:
Not sure if the validation for `currentIndex` will work with the string conversion.

I created a ticket for updating `isSafeDynamicKey` so that it handles all `PropertyKey` subtypes: https://github.com/MetaMask/utils/issues/200